packages/position/src/react/index.js

```diff
@@ -4,84 +4,152 @@ import ReactDOM from 'react-dom'
-  return <Position position={positionFns.aboveLeft} {...props} />
-}
+export const AboveLeft = React.forwardRef((props, ref) => {
+  return <Position position={positionFns.aboveLeft} ref={ref} {...props} />
+})
 AboveLeft.displayName = 'AboveLeft'
 
-export function AboveRight(props) {
-  return <Position position={positionFns.aboveRight} {...props} />
-}
+export const AboveRight = React.forwardRef((props, ref) => {
+  return <Position position={positionFns.aboveRight} ref={ref} {...props} />
+})
 AboveRight.displayName = 'AboveRight'
 
-export function Below(props) {
-  return <Position position={positionFns.below} {...props} />
-}
+export const Below = React.forwardRef((props, ref) => {
+  return <Position position={positionFns.below} ref={ref} {...props} />
+})
 Below.displayName = 'Below'
 
-export function BelowLeft(props) {
-  return <Position position={positionFns.belowLeft} {...props} />
-}
+export const BelowLeft = React.forwardRef((props, ref) => {
+  return <Position position={positionFns.belowLeft} ref={ref} {...props} />
+})
 BelowLeft.displayName = 'BelowLeft'
 
-export function BelowRight(props) {
-  return <Position position={positionFns.belowRight} {...props} />
-}
+export const BelowRight = React.forwardRef((props, ref) => {
+  return <Position position={positionFns.belowRight} ref={ref} {...props} />
+})
 BelowRight.displayName = 'BelowRight'
 
-export function RightOf(props) {
-  return <Position position={positionFns.rightOf} {...props} />
-}
+export const RightOf = React.forwardRef((props, ref) => {
+  return <Position position={positionFns.rightOf} ref={ref} {...props} />
+})
 RightOf.displayName = 'RightOf'
 
-export function LeftOf(props) {
-  return <Position position={positionFns.leftOf} {...props} />
-}
+export const LeftOf = React.forwardRef((props, ref) => {
+  return <Position position={positionFns.leftOf} ref={ref} {...props} />
+})
 LeftOf.displayName = 'LeftOf'
 
-function Position(props) {
-  const targetRef = React.useRef()
-  const showRef = React.useRef()
-  const child = React.Children.only(props.children)
+const Position = React.forwardRef((props, forwardedRef) => {
   const [style, setStyle] = React.useState({ position: 'absolute' })
 
-  React.useLayoutEffect(() => {
-    if (props.when && targetRef.current && showRef.current)
-      setStyle(props.position(targetRef.current).styleFor(showRef.current))
-  }, [props.when, targetRef, showRef, props])
-
-  const show =
-    props.when &&
-    React.cloneElement(props.show, {
-      ref: showRef,
-      style: {
-        ...child.props.style,
-        ...style
-      }
-    })
+  const ref = React.useRef()
+  React.useImperativeHandle(forwardedRef, () => ref.current)
+
+  const child = React.Children.only(props.children)
+  const tetheredRef = React.useRef()
+  const tetheredEl = React.cloneElement(props.show, {
+    ref: tetheredRef,
+    style: { ...child.props.style, ...style }
+  })
+
+  const updateStyle = React.useCallback(() => {
```

Comment:
Looks like these got missed.  Will you please name the Position component effect callbacks?

Reply:
ok i updated hooks but this one. because this is memoizing a function named `updateStyle` it seems redundant to also name the function. the name would probably just be `updateStyle` 🤷‍♂